src/main/java/seedu/savenus/model/food/FoodFilter.java

```diff
@@ -0,0 +1,108 @@
+package seedu.savenus.model.food;
+
+import static seedu.savenus.logic.parser.CliSyntax.FIELD_NAME_CATEGORY;
+import static seedu.savenus.logic.parser.CliSyntax.FIELD_NAME_DESCRIPTION;
+import static seedu.savenus.logic.parser.CliSyntax.FIELD_NAME_LOCATION;
+import static seedu.savenus.logic.parser.CliSyntax.FIELD_NAME_NAME;
+import static seedu.savenus.logic.parser.CliSyntax.FIELD_NAME_OPENING_HOURS;
+import static seedu.savenus.logic.parser.CliSyntax.FIELD_NAME_PRICE;
+import static seedu.savenus.logic.parser.CliSyntax.FIELD_NAME_RESTRICTIONS;
+import static seedu.savenus.logic.parser.CliSyntax.QUANTIFY_EQUALS_TO;
+import static seedu.savenus.logic.parser.CliSyntax.QUANTIFY_LESS_THAN;
+
+import java.util.List;
+import java.util.function.Predicate;
+
+/**
+ * Creates a new FoodFilter to filter all Foods not following the defined quantifiers.
+ */
+public class FoodFilter implements Predicate<Food> {
+
+    private List<String> fieldList;
+    private FieldComparator fieldComparator;
+
+    /**
+     * Constructs a Simple FoodFilter based on the fields given.
+     * @param fieldList the list of fields given.
+     */
+    public FoodFilter(List<String> fieldList) {
+        this.fieldList = fieldList;
+        this.fieldComparator = new FieldComparator();
+    }
+
+    @Override
+    public boolean test(Food food) {
+        return doesFoodPassTest(food);
+    }
+
+    /**
+     * Checks if the food fulfills certain quantifiers by testing it with all quantifiers.
+     * @param food the food to be tested.
+     * @return true if the food passes the tests. False if otherwise.
+     */
+    public boolean doesFoodPassTest(Food food) {
+        for (int i = 0; i < fieldList.size(); i = i + 3) {
+            String field = fieldList.get(i);
+            String quantifier = fieldList.get(i + 1);
+            String value = fieldList.get(i + 2);
+            Field dummyField = createDummyField(field, value);
```

Comment:
Maybe can parse the FieldList to create dummy field somewhere else first,
Because right now, for every food item, a new dummy field is created for each input field. 
So if the length of the food list is n, and the number of input fields to filter by m. the number of dummy field is created n x m times (Might be a lot of have many food). 
Instead, maybe can parse the input fields list first and map it to the corresponding dummy field then store it in a list. So the number of dummy fields created is just m.
Then loop through this list for each food.
^ I dunno whether I make sense, lemme know if I don't LOL